client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -57,22 +59,22 @@
     private final String scope;
     private final SynchronizerClientFactory clientFactory;
     private final Controller controller;
-    private final ConnectionFactory connectionFactory;
+    private final ConnectionPool connectionPool;
 
     public ReaderGroupManagerImpl(String scope, ClientConfig config, ConnectionFactory connectionFactory) {
         this.scope = scope;
         this.controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(config).build(),
                 connectionFactory.getInternalExecutor());
 
-        this.connectionFactory = connectionFactory;
+        this.connectionPool = new ConnectionPoolImpl(config, connectionFactory);
```

Comment:
Fixed

Reply:
I don't see any changes in `close`. Am I missing something?

Reply:
I changed it so that the clientFactory owns the pool and hence is responsible for it. 
I have made that more explicit by dropping the member variable.